### ISSUE TITLE
Mapserver quoting of identifiers for MySQL connections is incorrect

### ISSUE THREAD

**Issue by jlayt** [MapServer/MapServer#6612]
## Expected behavior and actual behavior.

MySQL/MariaDB database servers running in default mode interpret double-quotes as a string quote character and not as an identifier quote character as defined in standard ANSI SQL and used in other databases such as PostgreSQL.

Mapserver's OGR mapping of WFS FILTERs to SQL however uses the ANSI standard double-quoting for identifiers. Any MySQL server running in default mode thus fails to recognise the identifier in the generated SQL and returns 0 results. Only if the MySQL server is explicitly set to use ANSI mode will the Mapserver generated SQL return the expected results.

See https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_ansi_quotes for the official MySQL documentation.

The best solution is to use ` for MySQL identifiers instead as MySQL recognises this in both modes.

## Steps to reproduce the problem.

MySQL server running with default sql_modes set, i.e. no ANSI or ANSI_DOUBLEQUOTES set.

Mapserver Layer connection example:
```
  CONNECTION     'mysql:xxx_db,user=yyy,password=zzz,port=3306,tables=cor_tbl_spatial'
  DATA           'SELECT item,geometry,geomtype FROM cor_tbl_spatial' 
```
WFS Filter used in query:
```
  FILTER=<Filter><PropertyIsEqualTo><PropertyName>item</PropertyName><Literal>A1</Literal></PropertyIsEqualTo></Filter>
```
SQL generated by Mapserver as logged:
```
  SQL = SELECT id,geometry,geomtype FROM cor_tbl_spatial WHERE ((CAST("item" AS CHARACTER(4096)) = 'A1'))
```
This SQL returns 0 results, but SQL with "WHERE \`item\` = 'A1' " returns the expected results.

## Operating system

Debian 11 Bullseye

## MapServer version and installation method

Mapserver 7.6.2 from Debian 11 Bullseye packages

**Comment by jratike80** [comment]
Could you try to find the exact UTF-8 code for the quote character `? Is it this?
<html>
<body>


U+0060 | ` | 60 | GRAVE ACCENT
-- | -- | -- | --



</body>
</html>

**Comment by jlayt** [comment]
Yes, a number of sources confirm that it is the Grave Accent U+0060, which MySQL calls a 'backtick'.
